feat: extend the l1block contract to store historical block hash data

mikirov opened the pull request.

mikirov commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -74,5 +83,49 @@ contract L1Block is Semver {
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
+
+        _setL1BlockHash(_number, _hash);
```
Since we are only using this function once isn't it cheaper to save the function call and use the assembly here inline instead?

mikirov commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -74,5 +83,49 @@ contract L1Block is Semver {
+      }
+
+      if (_block < _first_number) {
+        revert L1BlockBlockHashIndexOutOfBound();
```
This might underflow if we deploy to a network that has less than 50k blocks already. This is not the case for optimism mainnet or testnets, however it's worth to bring awareness. Are we fine with that?

mikirov commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -74,5 +83,49 @@ contract L1Block is Semver {
+      bytes32 _hash;
+
+      assembly {
+        _first_number := add(sub(_last_number, BLOCK_HASHES_SIZE), 1)
```
This might underflow if we deploy to a network that has less than 50k blocks already. This is not the case for optimism mainnet or testnets, however it's worth to bring awareness. Are we fine with that?

failfmi commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -74,5 +85,36 @@ contract L1Block is Semver {
         batcherHash = _batcherHash;
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
+
```
space/tab size 2 does not match codebase 4

failfmi commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -74,5 +85,36 @@ contract L1Block is Semver {
+    /// @dev Requires the block number to be within the range of stored block hashes.
+    /// @param _number The number of the block for which to return the hash.
+    /// @return The hash of the L1 block.
+    function getL1BlockHash(uint64 _number) external view returns (bytes32) {
```
```solidity
function getL1BlockHash(uint64 _number) external view returns (bytes32 hash_) {
```

failfmi commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -74,5 +85,36 @@ contract L1Block is Semver {
+    /// @return The hash of the L1 block.
+    function getL1BlockHash(uint64 _number) external view returns (bytes32) {
+      uint64 _last_number = number;
+      bytes32 _hash;
```
no need to declare it, I think you can use `hash_` for consistency. Codebase uses this pattern for view functions, which return an argument.

failfmi commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -74,5 +85,36 @@ contract L1Block is Semver {
+    /// @param _number The number of the block for which to return the hash.
+    /// @return The hash of the L1 block.
+    function getL1BlockHash(uint64 _number) external view returns (bytes32) {
+      uint64 _last_number = number;
```
maybe rename to `lastNumber`?

failfmi commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -74,5 +85,36 @@ contract L1Block is Semver {
+
+      if (_last_number > BLOCK_HASHES_SIZE && _number <= _last_number - BLOCK_HASHES_SIZE) {
+        revert L1BlockHashNumberOutOfBounds();
+      }
```
What about:
```solidity
if (lastNumber - _number >= BLOCK_HASHES_SIZE) revert L1BlockHashNumberOutOfBounds();
```

failfmi commented on packages/contracts-bedrock/contracts/test/L1Block.t.sol:
```diff
@@ -11,6 +11,9 @@ contract L1BlockTest is CommonTest {
     L1Block lb;
     address depositor;
     bytes32 immutable NON_ZERO_HASH = keccak256(abi.encode(1));
+    uint16 public constant BLOCK_HASHES_SIZE = 50000;
```
Call contract in test rather than having it hard-coded in two places?

failfmi commented on packages/contracts-bedrock/contracts/test/L1Block.t.sol:
```diff
@@ -91,5 +95,150 @@ contract L1BlockTest is CommonTest {
+
+        /// @dev L1Block execution bytecode with `BLOCK_HASHES_SIZE` set to 50.
+        // solhint-disable-next-line
+        bytes memory l1BlockCodeBlockHashSize50 = hex"608060405234801561001057600080fd5b50600436106100df5760003560e01c80638381f58a1161008c5780639e8c4966116100665780639e8c4966146101ab578063b80777ea146101b4578063e591b282146101d4578063e81b2c6d1461021457600080fd5b80638381f58a146101735780638b239f73146101875780638e0b95361461019057600080fd5b806354fd4d50116100bd57806354fd4d50146101285780635cf249691461013 ...
```
Thoughts on switching the const with an immutable variable? Is the gas cost change significant?
Is there a problem testing it with > 50_000 records?

failfmi commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -14,6 +17,14 @@ contract L1Block is Semver {
 
+    /// @notice Storage slot with the address of the current block hashes offset.
+    /// @dev This is the keccak-256 hash of "l1block.block_hashes_storage_offset".
+    bytes32 private constant _BLOCK_HASHES_STORAGE_OFFSET =
```
`_BLOCK_HASHES_STORAGE_OFFSET` -> `BLOCK_HASHES_STORAGE_OFFSET`

Perseverance commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -74,5 +85,36 @@ contract L1Block is Semver {
+        _hash := sload(add(_BLOCK_HASHES_STORAGE_OFFSET, mod(_number, BLOCK_HASHES_SIZE)))
+      }
+
+      if (_hash == bytes32(0)) {
```
I think it is a good idea to add comment why _hash can ever be bytes32(0) given the two checks above

Perseverance commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -74,5 +85,36 @@ contract L1Block is Semver {
+        revert L1BlockHashNumberOutOfBounds();
+      }
+
+      if (_last_number > BLOCK_HASHES_SIZE && _number <= _last_number - BLOCK_HASHES_SIZE) {
```
Im not too sure why are we checking the first part? Block 50000 is long long gone

vladimir-trifonov replied:
I will reduce the number of inner cycle checks.

tynes commented:
Can you use this library? Copy in both the library implementation and the tests
https://github.com/0xJepsen/Solring/blob/master/src/RingLib.sol

We also need fuzz testing and to cover the test cases mentioned in the gov thread

tynes commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -14,6 +17,14 @@ contract L1Block is Semver {
+        0xdcb9291dfcbfcfdf70cc580ec1c1835768ebd0ed4a668ccb9e9e6167f533e655;
+
+    /// @notice The max count of block hashes to store.
+    uint16 public constant BLOCK_HASHES_SIZE = 50000;
```
This should be a power of 2

vladimir-trifonov replied:
We have benchmarked the two implementations and it turns out that they have the exact same gas profile.  I am genuinely curious to understand why you prefer to use the implementation that you have referenced instead of the current one?

Daniel-K-Ivanov commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -23,7 +23,7 @@ contract L1Block is Semver {
 
     /// @notice The max count of block hashes to store.
-    uint16 public constant BLOCK_HASHES_SIZE = 50000;
+    uint32 public constant BLOCK_HASHES_SIZE = 2**16; // 65536
```
Lets use the 2**16-1 (65635) value in order to fit into uint16 and not spill into uint32

Daniel-K-Ivanov commented on packages/contracts-bedrock/contracts/test/L1Block.t.sol:
```diff
@@ -91,5 +105,141 @@ contract L1BlockTest is CommonTest {
+        assertEq(lb.getL1BlockHash(3), keccak256(abi.encode(3)));
+    }
+
+    function test_setL1BlockValues_nonDepositor_reverts() external {
```
this could be changed to a fuzz test where we pass address as argument. Assume that it is not the depositor and expect reverts

Daniel-K-Ivanov commented on packages/contracts-bedrock/contracts/test/L1Block.t.sol:
```diff
@@ -91,5 +105,141 @@ contract L1BlockTest is CommonTest {
+        });
+    }
+
+    function test_getL1BlockHash_succeeds() external {
```
we could change this to a fuzz test where we have a BN provided as an argument and assume that the value is > n* BLOCK HASHES SIZE, where n is the number of times we want to "loop" the ring buffer.
We can add the `setL1BLockValues` in a for loop where we populate the block values starting from `value - BLOCK_HASHES_SIZE` until `value`. 

For each populated value we can check that the correct l1 block values have been populated.

Daniel-K-Ivanov replied:
having this as a fuzz test takes too much time

Daniel-K-Ivanov commented on packages/contracts-bedrock/contracts/test/L1Block.t.sol:
```diff
@@ -91,5 +105,141 @@ contract L1BlockTest is CommonTest {
+        lb.getL1BlockHash(2);
+    }
+
+    function test_getL1BlockHash_previousBlock_succeeds() external {
```
We can change this test to have bn passed as argument (fuzz test) and assume it is > 2. If that is the case - perform the setL1BlockValues twice and read the afterwards

Daniel-K-Ivanov commented on packages/contracts-bedrock/contracts/test/L1Block.t.sol:
```diff
@@ -91,5 +105,141 @@ contract L1BlockTest is CommonTest {
+        lb.getL1BlockHash(1);
+    }
+
+    function test_getL1BlockHash_hashNotSet_l1BlockHashNumberOutOfBounds_reverts() external {
```
This can be changed to a fuzz test where the bn passed from the fuzz test is assumed > 1. You would set the l1 block value for number 1 and asser that any other query for `bn` != 1 reverts

Daniel-K-Ivanov commented on packages/contracts-bedrock/contracts/test/L1Block.t.sol:
```diff
@@ -91,5 +105,141 @@ contract L1BlockTest is CommonTest {
+        // setup so we are expecting a revert here
+    }
+
+    function test_getL1BlockHash_lowerBound_l1BlockHashNumberOutOfBounds_reverts() external {
```
this can be changed to a fuzz test where you would set the l1blockvalues from 2 numbers provided.
The first one you would assume that it is uint256.max-BLOCK_HASHES_SIZE and the second one would be lower than the first one then query for `second_value` - expect that it will revert.

Daniel-K-Ivanov commented on packages/contracts-bedrock/contracts/test/L1Block.t.sol:
```diff
@@ -41,16 +47,24 @@ contract L1BlockTest is CommonTest {
         uint256 fo,
         uint256 fs
     ) external {
```
I think that we should not change this test. Its dedicated / focused solely on the setter for L1Blocks and individual getters for lastest bn. 

I think we should simply add the `assertEq(lb.getL1BlockHash(n)` and assert that it equals `h`

Daniel-K-Ivanov commented on packages/contracts-bedrock/contracts/test/L1Block.t.sol:
```diff
@@ -205,20 +240,19 @@ contract L1BlockTest is CommonTest {
     }
 
+    /// @dev Tests that `getL1BlockHash` returns correct values.
     function test_getL1BlockHash_succeeds() external {
```
nit: a better name for the test would be `test_getL1BlockHash_overwritten_ring_buffer_succeeds()`

vladimir-trifonov replied:
cool, I renamed it to test_getL1BlockHash_overwrittenRingBuffer_succeeds because the linter complains

tynes commented:
> We have benchmarked the two implementations and it turns out that they have the exact same gas profile. I am genuinely curious to understand why you prefer to use the implementation that you have referenced instead of the current one?

Its a portable library that has testing independent of the implementation in the portal. Would definitely prefer to use the library in this case so that we can have isolated test coverage of the feature

Daniel-K-Ivanov commented:
> Its a portable library that has testing independent of the implementation in the portal. Would definitely prefer to use the library in this case so that we can have isolated test coverage of the feature

@tynes Makes sense. We will update the PR with the usage of the library

tynes commented on packages/contracts-bedrock/contracts/test/L1Block.t.sol:
```diff
@@ -17,17 +19,30 @@ contract L1BlockTest is CommonTest {
-            _l1FeeOverhead: 2,
-            _l1FeeScalar: 3
-        });
+        lb.setL1BlockValues(n, t, b, h, s, bt, fo, fs);
```
Can you please undo this diff so args are explicit?

tynes commented on packages/contracts-bedrock/contracts/libraries/RingLib.sol:
```diff
@@ -0,0 +1,38 @@
+uint32 constant RING_SIZE = 65536;
+
+/// @title RingLib
+/// @notice Library for managing a ring buffer
```
Can you add back in the author tag with the original attribution?

tynes commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -38,6 +44,9 @@ contract L1Block is Semver {
+
     /// @custom:semver 1.0.1
     /// @notice Constructs the L1Block contract.
     constructor() Semver(1, 0, 1) {}
```
Can you bump the semver 1 minor version?

tynes commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -74,5 +83,30 @@ contract L1Block is Semver {
+    function getL1BlockHash(uint64 _number) external view returns (bytes32 hash_) {
+        uint64 lastNumber = number;
+
+        if (_number > lastNumber) {
```
Can we add some comments above each of these statement? Like: handle the case where the block is in the future or handle the case where the block is no longer in the ring buffer

tynes commented on packages/contracts-bedrock/contracts/L2/L1Block.sol:
```diff
@@ -2,6 +2,10 @@
+import { RingLib, RING_SIZE } from "../libraries/RingLib.sol";
+
+/// @notice Error code l1 block hash number out of bounds error.
+error L1BlockHashNumberOutOfBounds();
```
We are planning on updating to use custom errors in the future, do you mind using a require in the meantime for standardizing things?

tynes commented on packages/contracts-bedrock/contracts/test/RingLib.t.sol:
```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: Unlicense
```
Can you make this MIT?

vladimir-trifonov commented:
@tynes thanks for the comments :) all is fixed now

lastminutedev committed:
feat: extend the l1block contract to store historical block hash data

clabby committed:
Couple cleanups

clabby committed:
Start new `GameSolver` actor tests

clabby committed:
Handlers

Inphi committed:
feat(ctb): Specify L1 auth requirements

refcell committed:
fix(ctb): alphabet vm for output bisection

port alphabet changes to a new contract

bindings

fix: bindings

tynes committed:
contracts-bedrock: ERC721Bridge pausable

Enable pausability in the ERC721Bridge following the other bridges
that also received pausability.

Inphi committed:
complete owner configuration

Inphi committed:
update op-bindings

clabby committed:
Update to use `CommonBase`

tynes committed:
contracts-bedrock: more

tynes committed:
op-bindings: regenerate

tynes committed:
contracts-bedrock: generate snapshots

tynes committed:
contracts-bedrock: semver lock

refcell committed:
feat(op-challenger): Validate absolute prestate for split trace
providers.

fix(op-challenger): genesis output root validation for top of output bisection games

genesis output root test

fix(op-challenger): revert validation wiring

revert newline change

tynes committed:
contracts-bedrock: more cleanup

tynes committed:
contracts-bedrock: update snapshots

tynes committed:
contracts-bedrock: tests

tynes committed:
contracts-bedrock: fix bindings lockfile

tynes committed:
lint: fix

tynes committed:
contracts-bedrock: fix storage layout

tynes committed:
contracts-bedrock: delete storage layout file

ajsutton committed:
op-program: Verify loaded preimage matches requested key

Avoids errors in the preimage oracle (e.g. disk faults) from silently resulting in invalid execution.

clabby committed:
Merge pull request #8407 from ethereum-optimism/aj/output-cannon-1

Change output cannon game type to 1

EvanJRichard committed:
Merge pull request #8115 from ethereum-optimism/eip4844-e2e-minimal

EIP4844: Add single minimal e2e test.

Commit:
fix l2 block timestamp contraint for genesis block

dependabot[bot] committed:
build(deps-dev): bump eslint-config-prettier from 8.3.0 to 9.1.0

Bumps [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) from 8.3.0 to 9.1.0.
- [Changelog](https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/eslint-config-prettier/compare/v8.3.0...v9.1.0)

---
updated-dependencies:
- dependency-name: eslint-config-prettier
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 20.10.2 to 20.10.3

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.10.2 to 20.10.3.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump lint-staged from 15.1.0 to 15.2.0

Bumps [lint-staged](https://github.com/okonet/lint-staged) from 15.1.0 to 15.2.0.
- [Release notes](https://github.com/okonet/lint-staged/releases)
- [Changelog](https://github.com/lint-staged/lint-staged/blob/master/CHANGELOG.md)
- [Commits](https://github.com/okonet/lint-staged/compare/v15.1.0...v15.2.0)

---
updated-dependencies:
- dependency-name: lint-staged
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @sentry/node from 7.84.0 to 7.85.0

Bumps [@sentry/node](https://github.com/getsentry/sentry-javascript) from 7.84.0 to 7.85.0.
- [Release notes](https://github.com/getsentry/sentry-javascript/releases)
- [Changelog](https://github.com/getsentry/sentry-javascript/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/getsentry/sentry-javascript/compare/7.84.0...7.85.0)

---
updated-dependencies:
- dependency-name: "@sentry/node"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

tynes committed:
Merge pull request #8419 from ethereum-optimism/dependabot/npm_and_yarn/types/node-20.10.3

build(deps-dev): bump @types/node from 20.10.2 to 20.10.3

tynes committed:
contracts-bedrock: delete dead deployments

There is nothing in these deployment directories except
for the `.chainId` file. No need to keep them around if we are
not actually tracking deployments for these networks.

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 6.13.1 to 6.13.2

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 6.13.1 to 6.13.2.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.13.2/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump vitest from 0.34.1 to 1.0.1

Bumps [vitest](https://github.com/vitest-dev/vitest/tree/HEAD/packages/vitest) from 0.34.1 to 1.0.1.
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v1.0.1/packages/vitest)

---
updated-dependencies:
- dependency-name: vitest
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

tynes committed:
contracts-bedrock: flexible deploy script addresses

Allow for the addresses used in the deploy script to be defined
by a JSON file at the the path defined by the `ADDRESSES` environment
variable. This makes the deploy script easy to use when it comes to
composing with the `superchain-registry`. Just set the `ADDRESSES`
env var to the path on the local filesystem that points to the
`extra/addresses/<network>/<l2>.json` file. See the
[example](https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/extra/addresses/mainnet/op.json) here.

This gives a path forward for deprecating the concept of `deployments`
in the `contracts-bedrock` package. We really only use the `deployments`
directory for getting contract addresses. We want a single source of
truth for contract addresses which is `superchain-registry`. Using this
env var allows the deploy script to easily compose with
`superchain-registry`.

`ffi` and `jq` are used instead of `vm.readFile` to break out of the
security properties that foundry uses, we do not want to need to specify
all possible directories on the filesystem as readable in the
`foundry.toml`

tynes committed:
contracts-bedrock: remove dead imports

The `Safe` and `SafeProxyFactory` were imported twice.
This commit simply deletes one of the sets of imports so
the contracts are only imported once.

tynes committed:
Merge pull request #8402 from ethereum-optimism/ctb/l1xdm-sepolia-extended-pause

contracts-bedrock: L1CrossDomainMessenger sepolia

tynes committed:
Merge pull request #8424 from ethereum-optimism/dependabot/npm_and_yarn/vitest-1.0.1

build(deps-dev): bump vitest from 0.34.1 to 1.0.1

dependabot[bot] committed:
build(deps-dev): bump @vitest/coverage-istanbul from 0.34.6 to 1.0.1

Bumps [@vitest/coverage-istanbul](https://github.com/vitest-dev/vitest/tree/HEAD/packages/coverage-istanbul) from 0.34.6 to 1.0.1.
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v1.0.1/packages/coverage-istanbul)

---
updated-dependencies:
- dependency-name: "@vitest/coverage-istanbul"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

protolambda committed:
op-node: singular/span batch metric as eventvec with type label

protolambda committed:
Merge branch 'develop' into tip/spanbatch-logs-metrics

protolambda committed:
op-node: deduplicate rollup config override paths

ajsutton committed:
op-challenger: Add comment to avoid confusion about what GetBlockRange returns.

ajsutton committed:
op-e2e: Update output_cannon e2e test to execute a step and resolve the game.

ajsutton committed:
FaultDisputeGame: Remove TODO

The output bisection changes wound up in a new contract so no changes need to be made to the existing FaultDisputeGame contract and the TODO can be removed.

protolambda committed:
Merge pull request #8412 from testinprod-io/tip/spanbatch-logs-metrics

Add span batch logging & metrics

tynes committed:
Merge pull request #8423 from ethereum-optimism/dependabot/npm_and_yarn/typescript-eslint/parser-6.13.2

build(deps-dev): bump @typescript-eslint/parser from 6.13.1 to 6.13.2

tynes committed:
contracts-bedrock: add bool getter/setter to `Storage.sol`

Adds getters and setters for `bool` to the `Storage` library
as well as the `StorageSetter` contract. The tests happen through
the `StorageSetter` contract for simplicity. The `StorageSetter`
contract is important for the production 2 step upgrade process where
the storage migrations are decoupled from `initialize()`.

Also update the `StorageSetter` to use named return args per the latest
changes to the Solidity style guide.

clabby committed:
`LibPosition` updates + named local preimage keys

tynes committed:
contracts-bedrock: simplify storage setting in SuperchainConfig

Improve auditability + readability with some type safety.
Based on https://github.com/ethereum-optimism/optimism/pull/8435

tynes committed:
contracts-bedrock: deploy SuperchainConfig goerli

Initial deployment of `SuperchainConfig` to goerli.
Meant to be used as part of the "extended pause" contracts
release. Includes the deploy artifact for the deployment.
Only 1 deployment is needed per superchain target, meaning
that many L2 chains can use the same contract.

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 6.13.1 to 6.13.2.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.13.2/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

tynes committed:
contracts-bedrock: deploy `SuperchainConfig` to sepolia

Deploys the `SuperchainConfig` implementation to sepolia
and commits the deploy artifact. The contract was deployed
with the following command:

```bash

export IMPL_SALT=$(openssl rand -hex 32)

forge script scripts/Deploy.s.sol:Deploy \
  --sig 'deploySuperchainConfig()' --rpc-url $ETH_RPC_URL \
  --private-key $PRIVATE_KEY \
  --broadcast \
  --verify

```

tynes committed:
op-bindings: regenerate

tynes committed:
contracts-bedrock: regenerate snapshot

tynes committed:
contracts-bedrock: update semver lock

tynes committed:
Merge pull request #8422 from ethereum-optimism/dependabot/npm_and_yarn/lint-staged-15.2.0

build(deps-dev): bump lint-staged from 15.1.0 to 15.2.0

trianglesphere committed:
Fix lint

trianglesphere committed:
Merge branch 'develop' into rewrite-derivation-docs

trianglesphere committed:
Merge pull request #8130 from ethereum-optimism/seb/receipts-provider

op-service: Refactor `EthClient` with `ReceiptsProvider` abstraction

ajsutton committed:
Merge pull request #8432 from ethereum-optimism/aj/output_cannon_e2e

op-e2e: Update output_cannon e2e test to execute a step and resolve the game

trianglesphere committed:
Merge pull request #8430 from ethereum-optimism/aj/comments

op-challenger: Add comment to avoid confusion about what GetBlockRange returns

ImTei committed:
Log and set metrics after batch is derived without error

ajsutton committed:
op-e2e: Start migrating cannon e2e tests to output_cannon

Introduces a DisputeLastBlock helper to skip through the output root bisection so tests can focus on the bottom half of the game easily.
Tests are currently skipped as the pre-state is not being selected correctly.

sebastianst committed:
op-service: Use atomic int in receipts tests

numCalls are incremented concurrently, so access needs to be made
concurrency safe.
- testify.Mock doesn't hold a lock when calling a provided Run function.
- simple mock also has concurrent access.

tremarkley committed:
Merge pull request #8205 from ethereum-optimism/harry/faucet_drip_config

Update smallOpChainFaucetDripValue to be same as largeOpChainFaucetDr…

tynes committed:
Merge pull request #8441 from ethereum-optimism/seb/receipts-tests-atomic-numcalls

op-service: Use atomic int in receipts tests

maurelian committed:
Merge pull request #8427 from ethereum-optimism/ctb/cleanup-dead-deployments

contracts-bedrock: delete dead deployments

sukey2008 committed:
Fix several typos

Inphi committed:
fixup auth specs

refcell committed:
Merge pull request #8433 from ethereum-optimism/aj/remove-todo

FaultDisputeGame: Remove TODO

refcell committed:
Merge pull request #8405 from kylemasterd1/patch-1

Update pr-guidelines.md

Commit:
op-node: Add config flag to set max number of concurrent L1 RPC requests.
op-e2e: Add a default value to this flag in L1 setup.

Inphi committed:
fix merge conflict

clabby committed:
Bounded trace ancestor

clabby committed:
bindings + semver lock + lint

tynes committed:
Merge pull request #8444 from sukey2008/develop

Fix several typos in "ufm-test-services/README.md"

tynes committed:
Merge pull request #8426 from ethereum-optimism/dependabot/npm_and_yarn/vitest/coverage-istanbul-1.0.1

build(deps-dev): bump @vitest/coverage-istanbul from 0.34.6 to 1.0.1

protolambda committed:
op-node: add batch log meta data

EvanJRichard committed:
Merge pull request #8431 from ethereum-optimism/evan/configurable_max_rpc

Add config flag to set max number of concurrent L1 RPC requests.

dependabot[bot] committed:
build(deps-dev): bump vite from 5.0.4 to 5.0.5

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 5.0.4 to 5.0.5.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.0.5/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

refcell committed:
remove deploy script import

maurelian committed:
specs: Updates to superchain-configuration.md

refcell committed:
Merge pull request #8429 from ethereum-optimism/ctb/cleanup-imports-deploy

contracts-bedrock: remove dead imports

protolambda committed:
Merge pull request #8440 from testinprod-io/tip/fix-derived-batch-logging

op-node: Log and set metrics after batch is derived without error

refcell committed:
Merge pull request #8421 from ethereum-optimism/dependabot/npm_and_yarn/vite-5.0.5

build(deps-dev): bump vite from 5.0.4 to 5.0.5

refcell committed:
Merge pull request #8420 from ethereum-optimism/dependabot/npm_and_yarn/typescript-eslint/eslint-plugin-6.13.2

build(deps-dev): bump @typescript-eslint/eslint-plugin from 6.13.1 to 6.13.2

refcell committed:
Merge pull request #8399 from ethereum-optimism/ctb/deploy-l1xdm-goerli

contracts-bedrock: deploy extended pause L1CrossDomainMessenger

sukey2008 committed:
Fix one minor typo

sukey2008 committed:
Fix several typos

maurelian committed:
Merge pull request #8386 from ethereum-optimism/superchain-specs

Add spec for superchain-config with pausability

maurelian committed:
Merge pull request #8445 from sukey2008/develop

Several typos are fixed

Inphi committed:
Merge pull request #8361 from ethereum-optimism/inphi/auth

feat(ctb): Specify L1 auth requirements

maurelian committed:
contracts-bedrock: Add test_unpause_fullSystem_succeeds

dependabot[bot] committed:
build(deps-dev): bump @wagmi/core from 1.4.7 to 1.4.8

Bumps [@wagmi/core](https://github.com/wagmi-dev/wagmi/tree/HEAD/packages/core) from 1.4.7 to 1.4.8.
- [Release notes](https://github.com/wagmi-dev/wagmi/releases)
- [Changelog](https://github.com/wevm/wagmi/blob/main/packages/core/CHANGELOG.md)
- [Commits](https://github.com/wagmi-dev/wagmi/commits/@wagmi/core@1.4.8/packages/core)

---
updated-dependencies:
- dependency-name: "@wagmi/core"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump tsx from 4.6.1 to 4.6.2

Bumps [tsx](https://github.com/privatenumber/tsx) from 4.6.1 to 4.6.2.
- [Release notes](https://github.com/privatenumber/tsx/releases)
- [Changelog](https://github.com/privatenumber/tsx/blob/develop/release.config.cjs)
- [Commits](https://github.com/privatenumber/tsx/compare/v4.6.1...v4.6.2)

---
updated-dependencies:
- dependency-name: tsx
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

maurelian committed:
contracts-bedrock: make spacer variable non-public

maurelian committed:
contracts-bedrock: fix warning on _getSel, restrict to pure

pcw109550 committed:
specs: Add Span batch field protectedBits

pcw109550 committed:
op-node: Span batch protectedBits implementation

pcw109550 committed:
op-node: Span batch unit test upsert for protectedBits

pcw109550 committed:
op-node: Ensure unit tests to handle Span batches with unprotected txs

tynes committed:
Merge pull request #8450 from ethereum-optimism/dependabot/npm_and_yarn/tsx-4.6.2

build(deps-dev): bump tsx from 4.6.1 to 4.6.2

pcw109550 committed:
op-node: Fix span batch decode/encode logic error

maurelian committed:
Update OptimismPortal.sol

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

ajsutton committed:
op-e2e: Remove unused TempDir function

felipe-op committed:
feat(multisig-mon): integrate with 1pass (#8457)

* feat(multisig-mon): integrate with 1pass

* use logger instead of console.log

* lint

ajsutton committed:
op-e2e: Automatically close System at the end of the test

felipe-op committed:
feat(chainmon): add onepass to multisig-mon (#8461)

felipe-op committed:
revert forge-std unintentional changes from 51666c9ab95dcd379e01aeb7818fda6f74bbf38b (#8464)

trianglesphere committed:
op-batcher: Move config test from op-e2e to unit tests

tynes committed:
Merge pull request #8459 from ethereum-optimism/aj/use-normal-temp-dirs

op-e2e: Remove unused TempDir function

tynes committed:
Merge pull request #8447 from ethereum-optimism/dependabot/npm_and_yarn/wagmi/core-1.4.8

build(deps-dev): bump @wagmi/core from 1.4.7 to 1.4.8

tynes committed:
Merge pull request #8438 from ethereum-optimism/ctb/superchainconfig-sepolia-deploy

contracts-bedrock: deploy `SuperchainConfig` to sepolia

tynes committed:
Merge pull request #8418 from ethereum-optimism/dependabot/npm_and_yarn/eslint-config-prettier-9.1.0

build(deps-dev): bump eslint-config-prettier from 8.3.0 to 9.1.0

tynes committed:
Merge pull request #8439 from ethereum-optimism/aj/migrate-cannon-step-tests

op-e2e: Start migrating cannon e2e tests to output_cannon

dependabot[bot] committed:
build(deps-dev): bump eslint from 8.54.0 to 8.55.0

Bumps [eslint](https://github.com/eslint/eslint) from 8.54.0 to 8.55.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.54.0...v8.55.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

tynes committed:
contracts-bedrock: L2StandardBridge `initialize()`

Adds an `initialize()` function to the `L2StandardBridge`
implementation. This standardizes on having noop init functions
for L2 contracts. They are set in genesis state directly so the initialize
function should never execute. Keeping it there is more transparent.
Otherwise one may forget that it is initializable. The `_initialized`
storage slot MUST be set during L2 genesis generation.

ajsutton committed:
Merge pull request #8463 from ethereum-optimism/jg/batcher_config_test

op-batcher: Move config test from op-e2e to unit tests

maurelian committed:
Merge pull request #8455 from ethereum-optimism/ctb/pure-getsel

contracts-bedrock: fix warning on _getSel, restrict to pure

ajsutton committed:
Update check-changed to include op-preimage for cannon

tynes committed:
op-bindings: bump

tynes committed:
contracts-bedrock: update semver lock

tynes committed:
Merge pull request #8406 from testinprod-io/tip/delta-override

Add Delta override flag

tynes committed:
Merge pull request #8446 from ethereum-optimism/dependabot/npm_and_yarn/eslint-8.55.0

build(deps-dev): bump eslint from 8.54.0 to 8.55.0

tynes committed:
Merge pull request #8454 from ethereum-optimism/ctb/fix-spacer

contracts-bedrock: make spacer variable non-public

ajsutton committed:
Merge pull request #8460 from ethereum-optimism/aj/autoclose-system

op-e2e: Automatically close System at the end of the test

tynes committed:
Merge pull request #8437 from ethereum-optimism/ctb/deploy-superchainconfig-goerli

contracts-bedrock: deploy SuperchainConfig goerli

dependabot[bot] committed:
build(deps-dev): bump hardhat from 2.19.1 to 2.19.2

Bumps [hardhat](https://github.com/nomiclabs/hardhat) from 2.19.1 to 2.19.2.
- [Release notes](https://github.com/nomiclabs/hardhat/releases)
- [Commits](https://github.com/nomiclabs/hardhat/compare/hardhat@2.19.1...hardhat@2.19.2)

---
updated-dependencies:
- dependency-name: hardhat
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

trianglesphere committed:
Merge pull request #8342 from xenoliss/nit/op-node/calldata-source

nit(op-node): only store the needed rollup.Config fields in DataSource

tynes committed:
Merge pull request #8139 from roberto-bayardo/rewrite-derivation-docs

improve chain derivation documentation

felipe-op committed:
fix(multisig-mon): onepass to dockerfile

ajsutton committed:
op-challenger: Close the polling client.

ajsutton committed:
Merge pull request #8409 from ethereum-optimism/aj/oracle-validation

op-program: Verify loaded preimage matches requested key

hamdiallam committed:
Merge pull request #8160 from epociask/indexer.withdrawal_type_supplies

feat(indexer) - API type filtering (init, proven, finalized) for withdrawal values in `/supply` endpoint

hamdiallam committed:
Update indexer/docs/troubleshooting.md

ajsutton committed:
op-e2e: Use a fixed seed to make TestMixedDeposits repeatable.

ajsutton committed:
linkcheck: Exclude vitalik.ca

tynes committed:
Merge pull request #8471 from ethereum-optimism/aj/exclude-vitalik

linkcheck: Exclude vitalik.ca

tynes committed:
Merge pull request #8448 from ethereum-optimism/dependabot/npm_and_yarn/hardhat-2.19.2

build(deps-dev): bump hardhat from 2.19.1 to 2.19.2

tynes committed:
contracts-bedrock: update auth test

ajsutton committed:
op-e2e: Refactor dispute game helpers

Retrieve the various clients and endpoints required from the System rather than needing to pass them into multiple different methods.

maurelian committed:
Merge pull request #8365 from ethereum-optimism/ctb/erc721-pause

contracts-bedrock: ERC721Bridge pausable

ajsutton committed:
op-e2e: Add output_cannon version of TestCannonDefendStep

Currently ignored until contracts select the correct pre-state for steps.

mslipper committed:
Merge pull request #8470 from ethereum-optimism/aj/repeatable-test

op-e2e: Use a fixed seed to make TestMixedDeposits repeatable

hamdiallam committed:
Merge pull request #8084 from epociask/indexer.runbook

feat(Indexer) - Troubleshooting Runbook

pcw109550 committed:
specs: Fix Span batch EIP-155 description

ImTei committed:
Ensure to test all TX types including unprotected tx

ajsutton committed:
Merge pull request #8469 from ethereum-optimism/aj/close-poll-client

op-challenger: Close the polling client

protolambda committed:
span-batches: encode bitlists as big-endian integers, use standard Go big.Int encoding/decodig, update spec

taempark committed:
Merge pull request #43 from ethereum-optimism/span-batch-bitlists

span-batches: encode bitlists as big-endian integers, use standard Go…

tynes committed:
Merge pull request #8435 from ethereum-optimism/ctb/set-bool

contracts-bedrock: add bool getter/setter to `Storage.sol`

mslipper committed:
Merge pull request #8362 from ethereum-optimism/refcell/fix-alphabet-vm

fix(ctb): Output Bisection Alphabet VM

Pan-chao committed:
typo fixing

refcell committed:
feat(op-e2e): output alphabet e2e tests

fix(op-e2e): fix output alphabet test on refactor

Commit:
op-service: Add new eth client alternative PrefetchingEthClient, which proactively fills its cache with blocks, receipts, and txns.
Co-authored-by: Sebastian Stammler <seb@oplabs.co>

trianglesphere committed:
CI: Fix proxyd store tests

clabby committed:
Pos localization

maurelian committed:
contracts-bedrock: fix init tests

sebastianst committed:
Merge pull request #8191 from ethereum-optimism/prefetching_ethclient

Add `PrefetchingEthClient`, which builds a cache of ethclient data for callers.

maurelian committed:
bindings, snapshots, semver-lock

trianglesphere committed:
Merge pull request #8475 from Pan-chao/develop

specs: typo fix

trianglesphere committed:
Merge pull request #8458 from testinprod-io/tip/span-batch-protected-tx-patch

op-node: Span batch Unprotected Tx Handling + Logic Error Fix

ajsutton committed:
op-e2e: Rename l2NodeName for consistency.

sukey2008 committed:
Fix minor typos

maurelian committed:
Merge pull request #8465 from ethereum-optimism/ctb/l2-standard-bridge-init

contracts-bedrock: L2StandardBridge `initialize()`

sukey2008 committed:
Fix one typo

dependabot[bot] committed:
build(deps-dev): bump typescript from 5.3.2 to 5.3.3

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 5.3.2 to 5.3.3.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v5.3.2...v5.3.3)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

maurelian committed:
Merge pull request #8400 from ethereum-optimism/ctb/deploy-l1-standard-bridge-extended-pause

contracts-bedrock: Deploy L1StandardBridge extended pause

maurelian committed:
contracts-bedrock: Allow initializing system as paused

mslipper committed:
op-e2e: Use deterministic RNG in l2 batcher test

ajsutton committed:
Merge pull request #8472 from ethereum-optimism/aj/use-system

op-e2e: Refactor dispute game helpers

maurelian committed:
contracts-bedrock: fix version

ajsutton committed:
Merge pull request #8485 from ethereum-optimism/bugfix/determinstic-rand-batch-test

op-e2e: Use deterministic RNG in l2 batcher test

maurelian committed:
gas snapshot

mds1 committed:
chore: apply shellcheck auto fixes

To apply auto-fixes, the below command was run from the repo root:
find . -type f -name '*.sh' -exec sh -c 'shellcheck -f diff "$1" | patch "$1"' _ {} \;

mds1 committed:
chore: additional lint fixes suggested by code review

trianglesphere committed:
Merge pull request #8477 from ethereum-optimism/jg/proxyd_ci

CI: Fix proxyd store tests

tynes committed:
Merge pull request #8486 from ethereum-optimism/ctb/init-pausable

contracts-bedrock: Allow initializing system as paused

tynes committed:
Merge pull request #8482 from ethereum-optimism/dependabot/npm_and_yarn/typescript-5.3.3

build(deps-dev): bump typescript from 5.3.2 to 5.3.3

trianglesphere committed:
Merge pull request #8481 from sukey2008/develop

Fix several minor typos

refcell committed:
Merge pull request #8476 from ethereum-optimism/refcell/output-alphabet-e2e-tests

feat(op-e2e): Output Alphabet E2E Test

felipe-op committed:
fix(ufm): gracefully handle failed tx creation

maurelian committed:
Merge pull request #8479 from ethereum-optimism/ctb/superchain-config-set-bool-jm

contracts-bedrock: simplify storage setting in SuperchainConfig

refcell committed:
Merge pull request #8425 from ethereum-optimism/dependabot/npm_and_yarn/sentry/node-7.85.0

build(deps): bump @sentry/node from 7.84.0 to 7.85.0

refcell committed:
Merge pull request #8394 from ethereum-optimism/refcell/absolute-prestate-validation

feat(op-challenger): Output Bisection Game Genesis Output Root Fetching

refcell committed:
feat(op-challenger): prestate provider refactor

mslipper committed:
op-e2e: Add test utility to wait for balance changes

This PR adds a utility to wait for an address's balance to change. The utility checks for the balance change for two minutes, and returns the new value once it changes. It rreturns an error on timeout.

Some tests - particularly the ext-geth tests - are failing because balance changes that we expect to occur are not. My hunch is that this may be related to a race condition somewhere between the test code and op-geth, but I can't be sure. By waiting for the balance to change, we can eliminate spurious race conditions and see if this is related to some other underlying issue.

Akare123 committed:
Fix typos in README.md

Fixed a typo and recommend a change for better clarity:

1-Original: file which contains details of the L2 environment to bring up (including the
Suggested: file that contains details of the L2 environment to bring up (including the (I changed "which" to "that" for better flow)

2-Original: is included to help document an demonstrate the usage of the
Suggested: is included to help document and demonstrate the usage of the (I fixed a small typo, change "an" to "and")

felipe-op committed:
Merge pull request #8490 from ethereum-optimism/felipe/ufm-nil-tx

fix(ufm): gracefully handle failed tx creation

dependabot[bot] committed:
build(deps): bump actions/setup-python from 4 to 5

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 4 to 5.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

sebastianst committed:
Merge pull request #8492 from ethereum-optimism/bugfix/wait-for-balance-changes

op-e2e: Add test utility to wait for balance changes

mds1 committed:
chore: add quotes to BLOCK_SIGNER_ADDRESS

tynes committed:
Merge pull request #8494 from ethereum-optimism/dependabot/github_actions/actions/setup-python-5

build(deps): bump actions/setup-python from 4 to 5

Inphi committed:
feat(ctb): specify pausability of L1 contracts

tynes committed:
contracts-bedrock: deploy superchain config

Both the proxy and the latest implementation.
Deploy them to sepolia. The proxy was deployed with
the following command:

```bash
forge script scripts/Deploy.s.sol:Deploy \
  --sig 'deployERC1967Proxy(string)' --rpc-url $ETH_RPC_URL \
  --private-key $PRIVATE_KEY \
  --broadcast \
  --verify \
  SuperchainConfigProxy

```

maurelian committed:
Merge pull request #8497 from ethereum-optimism/ctb/sepolia-deployments

contracts-bedrock: deploy superchain config

tynes committed:
contracts-bedrock: add base sepolia deploy config

Sources the deploy config from https://github.com/base-org/contract-deployments/blob/main/sepolia/2023-09-26-deploy/deploy-config/deploy-config.json
We need to double check that this is up to date, ie it is possible that
some config has changed over time. The problem variables that we care
about from within this config are the constructor args to the
L2OutputOracle:

- cfg.l2OutputOracleSubmissionInterval
- cfg.l2BlockTime
- cfg.l2OutputOracleProposer
- cfg.l2OutputOracleChallenger
- cfg.finalizationPeriodSeconds

If any of these values changed, they should be caught by the state diff
tests written in `superchain-ops` for performing the upgrade.

This deploy config is required to deploy a new L2OutputOracle
implementation for Base Sepolia as part of the Extended Pause Upgrade.

clabby committed:
bindings

tynes committed:
contracts-bedrock: update deploy config

tynes committed:
contracts-bedrock: L2OutputOracle deployment

Deploys the latest implementation of the `L2OutputOracle` to sepolia
and commits the artifact.

The following script was used for the deployment.

```bash

export IMPL_SALT="op/acc"

forge script scripts/Deploy.s.sol:Deploy \
  --sig 'deployL2OutputOracle()' --rpc-url $ETH_RPC_URL \
  --private-key $PRIVATE_KEY \
  --broadcast \
  --verify
```

https://sepolia.etherscan.io/address/0x83aeb8b156cd90e64c702781c84a681dadb1dde2

clabby committed:
Merge pull request #8346 from ethereum-optimism/cl/ctb/output-bisection-test-refactor

feat(ctb): `OutputBisectionGame` test refactor + fixes

refcell committed:
Merge pull request #8491 from ethereum-optimism/refcell/prestate-provider-refactor

feat(op-challenger): Prestate Provider Refactor

tynes committed:
contracts-bedrock: zora-sepolia deploy config

Adds the deploy config for `zora-sepolia`. This was sourced from
conduit. This is required to deploy the new L2OutputOracle contract.

Related to https://github.com/ethereum-optimism/optimism/pull/8499

tynes committed:
contracts-bedrock: update zora sepolia config

Make it pass config checking

mslipper committed:
Merge pull request #8468 from ethereum-optimism/felipe/chainmon-add-onepass-to-dockerfile

fix(multisig-mon): add onepass cli to dockerfile

sukey2008 committed:
fix one typo

tynes committed:
contracts-bedrock: more fixes for config

tynes committed:
contracts-bedrock: fix pgn sepolia config

Updates the deploy config for pgn sepolia to
be conformant to the deploy config spec. Not sure why
CI is passing as the go code is supposed to check for
non compliant deploy config. It is likely because the solidity
and the go got out of sync. We could use a rethinking of how
we do deploy config. Ideally the entire thing can be simplified
now that deploy config contains op stack chain specific config,
superchain target specific config and also config values that
go into L1 and also config values that go into L2.

This change wouldn't be necessary if the lazy deploy config change
https://github.com/ethereum-optimism/optimism/issues/7711 was
implemented. It was attempted https://github.com/ethereum-optimism/optimism/pull/7776
but became stale and added a lot of overhead.

tynes committed:
Merge pull request #8500 from ethereum-optimism/ctb/sepolia-l2oo-deploy

contracts-bedrock: L2OutputOracle deployment

maurelian committed:
Update packages/contracts-bedrock/deploy-config/base-sepolia.json

tynes committed:
contracts-bedrock: deploy OptimismMintableERC20Factory sepolia

OP sepolia OptimismMintableERC20Factory deployment

Using the following command:
```

export IMPL_SALT="op/acc"
export ADDRESSES=/Users/tynes/Projects/github.com/ethereum-optimism/superchain-registry/superchain/extra/addresses/sepolia/op.json

forge script scripts/Deploy.s.sol:Deploy \
  --sig 'deployOptimismMintableERC20Factory()' --rpc-url $ETH_RPC_URL \
  --private-key $PRIVATE_KEY \
  --broadcast \
  --verify
```

mslipper committed:
proxyd: Delete bad test

This test was originally checking for a panic in the websocket code. However, we have since re-architected this code and have coverage for this issue elsewhere. Therefore, this test is now superfluous and can be removed. It causes a lot of spam in the logs, and makes debugging other tests harder.

maurelian committed:
Merge pull request #8501 from ethereum-optimism/ctb/zora-sepolia-deploy-config

contracts-bedrock: zora-sepolia deploy config

dependabot[bot] committed:
build(deps-dev): bump @vitest/coverage-istanbul from 1.0.1 to 1.0.2

Bumps [@vitest/coverage-istanbul](https://github.com/vitest-dev/vitest/tree/HEAD/packages/coverage-istanbul) from 1.0.1 to 1.0.2.
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v1.0.2/packages/coverage-istanbul)

---
updated-dependencies:
- dependency-name: "@vitest/coverage-istanbul"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @sentry/node from 7.85.0 to 7.86.0

Bumps [@sentry/node](https://github.com/getsentry/sentry-javascript) from 7.85.0 to 7.86.0.
- [Release notes](https://github.com/getsentry/sentry-javascript/releases)
- [Changelog](https://github.com/getsentry/sentry-javascript/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/getsentry/sentry-javascript/compare/7.85.0...7.86.0)

---
updated-dependencies:
- dependency-name: "@sentry/node"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 20.10.3 to 20.10.4

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.10.3 to 20.10.4.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

maurelian committed:
Merge pull request #8504 from ethereum-optimism/ctb/fix-pgn-sepolia

contracts-bedrock: fix pgn sepolia config

maurelian committed:
Merge pull request #8499 from ethereum-optimism/ctb/base-goerli-deploy-config

contracts-bedrock: add base sepolia deploy config

dependabot[bot] committed:
build(deps-dev): bump @wagmi/core from 1.4.8 to 1.4.10

Bumps [@wagmi/core](https://github.com/wagmi-dev/wagmi/tree/HEAD/packages/core) from 1.4.8 to 1.4.10.
- [Release notes](https://github.com/wagmi-dev/wagmi/releases)
- [Changelog](https://github.com/wevm/wagmi/blob/main/packages/core/CHANGELOG.md)
- [Commits](https://github.com/wagmi-dev/wagmi/commits/@wagmi/core@1.4.10/packages/core)

---
updated-dependencies:
- dependency-name: "@wagmi/core"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

ajsutton committed:
op-e2e: Adjust withdrawal_helper to reuse existing clients instead of dialling new ones.

The gethclient wrapper still needs to be created but it reuses the existing RPC client and doesn't have it's own Close function.

mslipper committed:
Merge pull request #8506 from ethereum-optimism/bugfix/proxyd-bad-test

proxyd: Delete bad test

trianglesphere committed:
Update op-e2e/external_geth/README.md

Co-authored-by: refcell.eth <abigger87@gmail.com>

trianglesphere committed:
Merge pull request #8503 from sukey2008/develop

fix one typo

maurelian committed:
Merge pull request #8401 from ethereum-optimism/ctb/deploy-portal-extended-pause

contracts-bedrock: deploy OptimismPortal extended pause

refcell committed:
Merge pull request #8511 from ethereum-optimism/dependabot/npm_and_yarn/wagmi/core-1.4.10

build(deps-dev): bump @wagmi/core from 1.4.8 to 1.4.10

tynes committed:
Merge pull request #8505 from ethereum-optimism/ctb/factory-sepolia

contracts-bedrock: deploy OptimismMintableERC20Factory sepolia

tynes committed:
Merge pull request #8510 from ethereum-optimism/dependabot/npm_and_yarn/types/node-20.10.4

build(deps-dev): bump @types/node from 20.10.3 to 20.10.4

refcell committed:
Merge pull request #8509 from ethereum-optimism/dependabot/npm_and_yarn/sentry/node-7.86.0

build(deps): bump @sentry/node from 7.85.0 to 7.86.0

tynes committed:
Merge pull request #8507 from ethereum-optimism/dependabot/npm_and_yarn/vitest/coverage-istanbul-1.0.2

build(deps-dev): bump @vitest/coverage-istanbul from 1.0.1 to 1.0.2

tynes committed:
Merge pull request #8498 from ethereum-optimism/inphi/auth-pause

feat(ctb): specify pausability of L1 contracts

trianglesphere committed:
Merge pull request #8493 from Akare123/patch-1

Fix typos in README.md

ajsutton committed:
Merge pull request #8513 from ethereum-optimism/aj/reuse-client

op-e2e: Adjust withdrawal_helper to reuse existing clients instead of dialling new ones

tynes committed:
Merge pull request #8487 from ethereum-optimism/chore/shellcheck-setup

chore: apply shellcheck auto fixes

mslipper committed:
devnet: Convert devnet-allocs to Anvil

Converts devnet-allocs to use Anvil rather than dev mode Geth. This should both be faster than dev mode geth, as well as more reliable since there's an issue with devnet geth hanging during builds.

tynes committed:
contracts-bedrock: modularize addresses abstraction

dependabot[bot] committed:
build(deps-dev): bump vite from 5.0.5 to 5.0.6

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 5.0.5 to 5.0.6.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.0.6/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mslipper committed:
Convert balances to decimal rather than hex

tynes committed:
contracts-bedrock: lint

tynes committed:
contracts-bedrock: fix base sepolia config

Makes the L1 starting blocknumber concrete.
See commit message in https://github.com/ethereum-optimism/optimism/pull/8515
for an explanation.

The value in the config comes from here:
https://sepolia.etherscan.io/address/0x84457ca9D0163FbC4bbfe4Dfbb20ba46e48DF254#readProxyContract

The contract address sourced from here:
https://github.com/base-org/contract-deployments/blob/910151eb2eed3542efbb4ccf7e38d1ed3b79d8f5/sepolia/2023-09-26-deploy/deployed/addresses.json#L10C27-L10C69

tynes committed:
contracts-bedrock: update sepolia deploy config

See https://github.com/ethereum-optimism/optimism/pull/8515 for
explanation. Also related to https://github.com/ethereum-optimism/optimism/pull/8519.

Value fetched from:
https://sepolia.etherscan.io/address/0xD5bAc3152ffC25318F848B3DD5dA6C85171BaEEe#readProxyContract

Address sourced from:
https://github.com/ethereum-optimism/superchain-registry/blob/fb3433d89a72df5809255c065b4783fa393608d8/superchain/extra/addresses/sepolia/pgn.json#L6C27-L6C69

trianglesphere committed:
Merge pull request #8517 from ethereum-optimism/feat/anvil-devnet-allocs

devnet: Convert devnet-allocs to Anvil

maurelian committed:
Merge pull request #8520 from ethereum-optimism/ctb/pgn-sepolia-deploy-config

contracts-bedrock: update pgn sepolia deploy config

maurelian committed:
contracts-bedrock: Fix README

maurelian committed:
Merge pull request #8428 from ethereum-optimism/ctb/flexible-deploy-addresses

contracts-bedrock: flexible deploy script addresses

tynes committed:
Merge pull request #8519 from ethereum-optimism/ctb/sepolia-base-concrete-config

contracts-bedrock: fix base sepolia config

tynes committed:
Merge pull request #8483 from ethereum-optimism/dependabot/npm_and_yarn/vite-5.0.6

build(deps-dev): bump vite from 5.0.5 to 5.0.6

lastminutedev committed:
fix: fix build

lastminutedev committed:
test: fix coderabit comment

lastminutedev committed:
test: fix incorrect coverage failings
